conans/client/installer.py

```diff
@@ -536,13 +539,28 @@ def _handle_node_cache(self, node, processed_package_references, remotes, pkg_la
             if pkg_layout.reference != pref:
                 self._cache.assign_prev(pkg_layout, ConanReference(pref))
 
-            package_folder = pkg_layout.package()
+            if os.path.exists(pkg_layout.post_package()):
+                output.info('Using package install folder')
+                package_folder = pkg_layout.post_package()
+            else:
+                package_folder = pkg_layout.package()
             assert os.path.isdir(package_folder), ("Package '%s' folder must exist: %s\n"
                                                    % (str(pref), package_folder))
             # Call the info method
             self._call_package_info(conanfile, package_folder, ref=pref.ref, is_editable=False)
             self._recorder.package_cpp_info(pref, conanfile.cpp_info)
 
+    def _call_post_package(self, conanfile, output, pkg_layout):
+        # FIXME: better name? "install" might be confusing with the old install_folder mechanism
+        # FIXME: Maybe it is better to remove first the install folder here in develop2
+        # TODO: dirty?
```

Comment:
Yes, seems "post_package" needs its own dirty control check, as it can be interrupted, fail, etc.